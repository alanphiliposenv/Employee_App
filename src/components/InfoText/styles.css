.info-text {
    width: calc(100%/5);
}

.info-text-label {
  font: normal normal 400 16px/22px Nunito;
  letter-spacing: 0px;
  color: #8b8d9d;
}

.info-text-value {
  text-align: left;
  font: normal normal 400 16px/22px Nunito;
  letter-spacing: 0px;
  color: #434552;
}
