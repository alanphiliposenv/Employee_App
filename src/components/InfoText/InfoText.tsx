import { FC } from 'react';
import Status from '../Status/Status';
import './styles.css';

export type InfoTextPropTypes = {
  type?: 'address' | 'text' | 'status';
  label?: string;
  value?: any;
};

const InfoText: FC<InfoTextPropTypes> = ({ label, type, value }) => {
  return (
    <div className='info-text'>
      <div className='info-text-label' data-testid='info-text-label'>
        {label}
      </div>
      {type === 'status' && <Status status={value} />}
      {type === 'text' && <div className='info-text-value'>{value}</div>}
      {type === 'address' && (
        <div className='info-text-value'>{`${value.address_line_1}, ${value.address_line_2}, ${value.city}, ${value.state}, ${value.country} - ${value.pincode}`}</div>
      )}
    </div>
  );
};

export default InfoText;
