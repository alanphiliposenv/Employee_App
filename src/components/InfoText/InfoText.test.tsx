import { render } from '@testing-library/react';
import InfoText, { InfoTextPropTypes } from './InfoText';

describe('test for info text', () => {
  test('snapshot redered as text', () => {
    const InfoTextProps: InfoTextPropTypes = {
      type: 'text',
      value: 'text-type'
    };

    const element = render(<InfoText {...InfoTextProps} />);

    expect(element).toMatchSnapshot();
  });
  test('snapshot redered as status', () => {
    const InfoTextProps: InfoTextPropTypes = {
      type: 'address',
      value: {
        address_line_1: 'line1',
        address_line_2: 'line2',
        city: 'city',
        state: 'state',
        country: 'country',
        pincode: 'pincode'
      }
    };

    const element = render(<InfoText {...InfoTextProps} />);

    expect(element).toMatchSnapshot();
  });
  test('snapshot redered as status', () => {
    const InfoTextProps: InfoTextPropTypes = {
      type: 'status',
      value: 'active'
    };

    const element = render(<InfoText {...InfoTextProps} />);

    expect(element).toMatchSnapshot();
  });
});
