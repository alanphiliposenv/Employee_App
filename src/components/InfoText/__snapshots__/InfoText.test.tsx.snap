// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for info text snapshot redered as status 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="info-text"
      >
        <div
          class="info-text-label"
          data-testid="info-text-label"
        />
        <div
          class="info-text-value"
        >
          line1, line2, city, state, country - pincode
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="info-text"
    >
      <div
        class="info-text-label"
        data-testid="info-text-label"
      />
      <div
        class="info-text-value"
      >
        line1, line2, city, state, country - pincode
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test for info text snapshot redered as status 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="info-text"
      >
        <div
          class="info-text-label"
          data-testid="info-text-label"
        />
        <div
          class="status active"
        >
          Active
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="info-text"
    >
      <div
        class="info-text-label"
        data-testid="info-text-label"
      />
      <div
        class="status active"
      >
        Active
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test for info text snapshot redered as text 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="info-text"
      >
        <div
          class="info-text-label"
          data-testid="info-text-label"
        />
        <div
          class="info-text-value"
        >
          text-type
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="info-text"
    >
      <div
        class="info-text-label"
        data-testid="info-text-label"
      />
      <div
        class="info-text-value"
      >
        text-type
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
